lfs/scanner.go

```diff
@@ -187,6 +195,12 @@ func revListShas(refLeft, refRight string, opt ScanRefsOptions) (chan string, er
 		if refRight != "" && !z40.MatchString(refRight) {
 			refArgs = append(refArgs, refRight)
 		}
+	case ScanAll:
+		refArgs = append(refArgs, "--all")
+	case ScanDeltaToRemotes:
+		refArgs = append(refArgs, refLeft, "--not", "--remotes")
```

Comment:
> But since we only have one git-lfs server and single lfs.url configuration option, it is safe to assume that the same git lfs server is used for all remotes. So, we can do a better jobs isolating only lfs objects introduced locally.

Not necessarily. You can also configure an lfs server at `remote.{name}.lfsurl`. Ideally, each git host has its own lfs host too that's automatically configured. It should be easy to push and pull between different git hosts.

There is one big issue with this right now. Git doesn't tell us what remote we should be using in the smudge or post-checkout hooks. So pulling from a non-origin remote doesn't quite work.